src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_CAPACITY = 16;
+    public static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size = 0;
+    private int threshold;
+    private Node<K, V> [] table;
+
+    class Node<K, V> {
+        private final int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = hash(key) % table.length;
+        Node<K, V> current = table[index];
+        if (current == null) {
+            table[index] = new Node<>(hash(key), key, value, null);
+            size++;
+            checkResize();
+            return;
+        }
+        while (current != null) {
+            if (current.key != null && current.key.equals(key)) {
+                current.value = value;
+                return;
+            } else if (current.key == null && key == null) {
+                current.value = value;
+                return;
+            }
+            if (current.next == null) {
+                break;
+            }
+            current = current.next;
+        }
+        current.next = new Node<>(hash(key), key, value, null);
+        size++;
+        checkResize();
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key) % table.length;
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if (current.key != null && current.key.equals(key)) {
+                return current.value;
+            } else if (current.key == null && key == null) {
+                return current.value;
+            }
+            if (current.next == null) {
+                break;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
```

Comment:
You have a bug in the `hash` method. You are always applying bitwise AND operation `&` with `DEFAULT_CAPACITY - 1`. This operation is used to limit the hash code to the current capacity of the table. But you are always using `DEFAULT_CAPACITY` instead of the current capacity `table.length`. This will cause problems when the table is resized.